provider/packet/environ.go

```diff
@@ -316,6 +366,114 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	}, nil
 }
 
+// supportedInstanceTypes returns the instance types supported by packet.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	plans, _, err := e.packetClient.Plans.List(new(packngo.ListOptions))
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving supported instance types")
+	}
+
+	var instTypes []instances.InstanceType
+
+nextPlan:
+	for _, plan := range plans {
+		if !validPlan(plan) {
+			continue
+		}
+
+		// TODO: Only work with this plan type for now to avoid
+		// spinning up more expensive instances while testing.
+		if plan.Name != "t1.small.x86" {
+			continue
+		}
+
+		var instArch string
+		switch {
+		case strings.HasSuffix(plan.Name, ".x86"):
+			instArch = arch.AMD64
+		case strings.HasSuffix(plan.Name, ".arm"):
+			instArch = arch.AMD64
+		default:
+			continue nextPlan
+
+		}
+
+		mem, err := parseMemValue(plan.Specs.Memory.Total)
+		if err != nil {
+			continue
+		}
+
+		instTypes = append(instTypes,
+			instances.InstanceType{
+				Id:       plan.ID,
+				Name:     plan.Name,
+				CpuCores: uint64(plan.Specs.Cpus[0].Count),
+				Mem:      mem,
+				Arches:   []string{instArch},
+				// Scale per hour costs so they can be represented as an integer for sorting purposes.
+				Cost: uint64(plan.Pricing.Hour * 1000.0),
+				// TODO: returned by packet's API but not exposed by the packngo client
+				// Deprecated: plan.Legacy,
```

Comment:
@displague Can the client be updated to expose this information? It would allow juju to make a more intelligent decision when it selects the instance to use (and also warn the user if it selects a deprecated instance due to constraints etc.).

Reply:
https://github.com/packethost/packngo/issues/232